modApiServer/test/org/aion/api/server/ApiAionTests.java

```diff
@@ -0,0 +1,398 @@
+            pendingRcvdFlag = true;
+        }
+
+        @Override
+        protected void pendingTxUpdate(ITxReceipt _txRcpt, EventTx.STATE _state) {
+            pendingUpdateFlag = true;
+        }
+
+        public boolean allFlagsSet() {
+            return (onBlockFlag && pendingRcvdFlag && pendingUpdateFlag);
+        }
+
+
+        public ApiAionImpl(AionImpl impl) {
+            super(impl);
+            onBlockFlag = false;
+            pendingRcvdFlag = false;
+            pendingUpdateFlag = false;
+        }
+
+        public void addEvents() {
+            EventTx pendingRcvd = new EventTx(EventTx.CALLBACK.PENDINGTXRECEIVED0);
+            AionTransaction tx = new AionTransaction(null);
+            List l1 = new ArrayList<ITransaction>();
+            l1.add(tx);
+            l1.add(tx);
+            l1.add(tx);
+            pendingRcvd.setFuncArgs(Collections.singletonList(l1));
+
+            ees.add(pendingRcvd);
+
+            EventTx pendingUpdate = new EventTx(EventTx.CALLBACK.PENDINGTXUPDATE0);
+            List l2 = new ArrayList<>();
+            l2.add(new AionTxReceipt());
+            l2.add(-1);
+            pendingUpdate.setFuncArgs(l2);
+
+            ees.add(pendingUpdate);
+
+            EventBlock evBlock = new EventBlock(EventBlock.CALLBACK.ONBLOCK0);
+            AionBlockSummary abs = new AionBlockSummary(null, null, null, null);
+            evBlock.setFuncArgs(Collections.singletonList(abs));
+
+            ees.add(evBlock);
+
+            //provokes exception in EpApi.run()
+            ees.add(new EventBlock(EventBlock.CALLBACK.ONBLOCK0));
+            ees.add(new EventDummy());
+        }
+
+    }
+
+    @Test
+    public void TestCreate() {
+        System.out.println("run TestCreate.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        System.out.println("API Version = " + api.getApiVersion());
+        assertNotNull(api.getInstalledFltrs());
+    }
+
+    @Test
+    public void TestInitNrgOracle() {
+        System.out.println("run TestInitNrgOracle.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        api.initNrgOracle(impl);
+        assertNotNull(api.getNrgOracle());
+        // Initing a second time should not create a new NrgOracle
+        api.initNrgOracle(impl);
+        assertNotNull(api.getNrgOracle());
+    }
+
+    @Test
+    public void TestStartES() {
+        System.out.println("run TestStartES.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        api.startES("thName");
+        api.addEvents();
+        try
+        {
+            Thread.sleep(5000);
+        }
+        catch(InterruptedException ex) {}
+        api.shutDownES();
+        assertTrue(api.allFlagsSet());
+    }
+
+    @Test
+    public void TestGetBlock() {
```

Comment:
Probably not worth splitting into 3 separate tests since the cases are so small, but it would be valuable to break up the 3 cases with comments -> using -1, 0 and then n > 0